android/src/main/kotlin/carnegietechnologies/gallery_saver/GallerySaver.kt

```diff
@@ -26,109 +29,72 @@ class GallerySaver internal constructor(private val activity: Activity) : Plugin
      *
      * @param methodCall - method call
      * @param result     - result to be set when saving operation finishes
-     * @param isImage    - tells if image or video should be saved
+     * @param mediaType    - media type
      */
-    internal fun saveFile(methodCall: MethodCall, result: MethodChannel.Result, isImage: Boolean) {
-        if (!setPendingMethodCallAndResult(methodCall, result)) {
-            finishWithError()
-            return
-        }
-
-        if (ActivityCompat.checkSelfPermission(activity, Manifest.permission.WRITE_EXTERNAL_STORAGE)
-                != PackageManager.PERMISSION_GRANTED) {
-            ActivityCompat.requestPermissions(activity,
-                    arrayOf(Manifest.permission.WRITE_EXTERNAL_STORAGE),
-                    REQUEST_EXTERNAL_IMAGE_STORAGE_PERMISSION)
-            this.methodCall = methodCall
-            this.pendingResult = result
-            this.isImage = isImage
-            return
-        }
-        if (isImage) {
-            saveImage(methodCall)
+    internal fun checkPermissionAndSaveFile(
+        methodCall: MethodCall,
+        result: MethodChannel.Result,
+        mediaType: MediaType
+    ) {
+        filePath = methodCall.argument<Any>(KEY_PATH)?.toString() ?: ""
+        albumName = methodCall.argument<Any>(KEY_ALBUM_NAME)?.toString() ?: ""
+        this.mediaType = mediaType
+        this.pendingResult = result
+
+        if (isWritePermissionGranted()) {
+            saveMediaFile()
         } else {
-            saveVideo(methodCall)
-        }
-    }
+            ActivityCompat.requestPermissions(
+                activity,
+                arrayOf(Manifest.permission.WRITE_EXTERNAL_STORAGE),
+                REQUEST_EXTERNAL_IMAGE_STORAGE_PERMISSION
+            )
 
-    /**
-     * Saves video with provided name and description to device
-     *
-     * @param methodCall - method call
-     */
-    private fun saveVideo(methodCall: MethodCall) {
-        val tempPath = if (methodCall.argument<Any>(KEY_PATH) == null)
-            ""
-        else
-            methodCall.argument<Any>(KEY_PATH)!!.toString()
-
-        uiScope.launch {
-            val success = async(Dispatchers.IO) {
-                FileUtils.insertVideo(activity.contentResolver, tempPath)
-            }
-            success.await()
-            finishWithSuccess()
         }
     }
 
+    private fun isWritePermissionGranted(): Boolean {
+        return PackageManager.PERMISSION_GRANTED ==
+                ActivityCompat.checkSelfPermission(
+                    activity, Manifest.permission.WRITE_EXTERNAL_STORAGE
+                )
+    }
 
     /**
-     * Saves image with provided image and description to device
+     * Saves media file
```

Comment:
So saveMediaFile(), according to javadoc 'Saves media file'. Who would have known??? :)

Either elaborate on what this method is doing or remove redundant javadoc.